### PR TITLE
Eng 10873

### PR DESCRIPTION
**Pull request by hhakimi53**
Updates to fix issue where Decimal (BigDecimal) number was getting transmitted incorrectly from C# client. It was happening due to bit calculation for certain range of BigDecimal numbers not happening appropriately. Changes include:
1 - Fixes for bit shift issue when fetching bytes  in third party BigInteger logic. Fixes based on https://github.com/bazzilic/BigInteger/commit/eff19aa58f3166d0148eedad17a3696c528dbf71
2 - Incorporate other fixes related to buffer overflows in BigInteger logic from https://github.com/bazzilic/BigInteger and code clean-up
3 - Update to kit packaging script and VS project cleanup/fixes for path

### CODE REVIEW

File: VoltDB.Data.Client/Types/ThirdParty.BigInteger.cs [hunk 1]
```diff
@@ -2694,19 +2619,26 @@ public byte[] getBytes()
...
-
             int pos = 0;
             uint tempVal, val = data[dataLength - 1];
 
             if ((tempVal = (val >> 24 & 0xFF)) != 0)
                 result[pos++] = (byte)tempVal;
+
             if ((tempVal = (val >> 16 & 0xFF)) != 0)
                 result[pos++] = (byte)tempVal;
+            else if (pos > 0)
```
Comment (akhanzode) [inline VoltDB.Data.Client/Types/ThirdParty.BigInteger.cs]: What do you supposed to put in result[pos] if you are in else? 0?

**Comment by akhanzode** [comment]
Looks good to me.

File: VoltDB.Data.Client/Types/ThirdParty.BigInteger.cs (hunk 1 above)
Comment (hhakimi53) [inline VoltDB.Data.Client/Types/ThirdParty.BigInteger.cs]: Yes, if in else the result[pos] should be 0. Byte[] when created, have all values in [] initialized to zero. So don't need to explicitly assign 0 for skipped positions.

**Comment by hhakimi53** [comment]
Thanks Anish for reviewing the changes.

**Pull request merged**